crypto/cms/cms_lib.c

```diff
@@ -74,11 +74,7 @@ CMS_ContentInfo *CMS_ContentInfo_new(void)
 
 void CMS_ContentInfo_free(CMS_ContentInfo *cms)
 {
-    if (cms != NULL) {
```

Comment:
I dont think it is a good idea to remove the NULL check.

Reply:
the default implementation has no NULL check either, but
maybe I should use `IMPLEMENT_ASN1_ALLOC_FUNCTIONS(CMS_ContentInfo)`
instead of implementing it here, possibly different.

Reply:
I was thinking it is a public function.. So it should do the NULL check.

Reply:
yes the NULL-check is in the `ASN1_item_free` below.